src/components/ChooseSubmodule/ChooseSubmodule.tsx

```diff
@@ -0,0 +1,98 @@
+import { ArrowBackIcon, ChevronDownIcon } from "@chakra-ui/icons";
+import { Button, Flex, useDisclosure, useToken } from "@chakra-ui/react";
+import type { Assembly } from "jsii-reflect";
+import { FunctionComponent, useCallback, useMemo, useState } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+import { useQueryParams } from "../../hooks/useQueryParams";
+import { SearchPopover } from "./SearchPopover";
+
+export interface ChooseSubmoduleProps {
+  assembly?: Assembly;
+}
+
+export const ChooseSubmodule: FunctionComponent<ChooseSubmoduleProps> = ({
+  assembly,
+}) => {
+  const { pathname } = useLocation();
+  const { push } = useHistory();
+  const query = useQueryParams();
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const [borderColor, textColor] = useToken("colors", ["gray.100", "gray.800"]);
+  const btnHeight = useToken("space", "16");
+
+  const currentSubmodule = query.get("submodule");
+  const submoduleText = currentSubmodule ?? "Submodules";
+
+  const [filter, setFilter] = useState("");
+
+  const onGoBack = () => {
+    const lang = query.get("lang");
+    push(`${pathname}${lang ? `?lang=${lang}` : ""}`);
+  };
+
+  const getUrl = useCallback(
+    (submoduleName: string) => {
+      const params = new URLSearchParams(query.toString());
+      params.set("submodule", submoduleName);
+      return `${pathname}?${params}`;
+    },
+    [pathname, query]
+  );
+
+  const submodules = useMemo(() => {
+    let results = assembly?.submodules ?? [];
+
+    if (filter) {
+      results = results.filter(({ name }) =>
+        name.toLowerCase().includes(filter.toLowerCase())
+      );
+    }
+
+    return results.map(({ name }) => ({
+      name,
+      to: getUrl(name),
+    }));
+  }, [assembly?.submodules, filter, getUrl]);
```

Comment:
Can we compute this all and pass as props to a simple presentational instead of having this in the same function as the presentational stuff?

Reply:
I don't really know that it makes sense to break this component apart any more than it already is

Reply:
yeah I'm fine with it on further inspection.